test/wdio/slot-dynamic-name-change/cmp.test.tsx

```diff
@@ -0,0 +1,54 @@
+import { Fragment, h } from '@stencil/core';
+import { render } from '@wdio/browser-runner/stencil';
+
+/**
+ * Tests the case where a `slot` element in a component has its
+ * `name` attribute changed dynamically via a property.
+ */
+describe('slot dynamic name change', () => {
+  beforeEach(async () => {
+    render({
+      template: () => (
+        <>
+          <slot-dynamic-name-change-shadow>
+            <p slot="greeting">Hello</p>
+            <p slot="farewell">Goodbye</p>
+          </slot-dynamic-name-change-shadow>
+          <slot-dynamic-name-change-scoped>
+            <p slot="greeting">Hello</p>
+            <p slot="farewell">Goodbye</p>
+          </slot-dynamic-name-change-scoped>
+
+          <button>Toggle slot name</button>
+        </>
+      ),
+    });
+
+    document.querySelector('button').addEventListener('click', () => {
+      document.querySelector('slot-dynamic-name-change-shadow').setAttribute('slot-name', 'farewell');
+      document.querySelector('slot-dynamic-name-change-scoped').setAttribute('slot-name', 'farewell');
+    });
```

Comment:
General question - should we be calling `removeEventListener` for these in an `afterEach`?

Reply:
we could, but this wouldn't change anything as we do page reloads between tests so these event listeners are being remove anyway.